lib/vector-core/src/event/finalization.rs

```diff
@@ -191,6 +191,13 @@ impl BatchStatusReceiver {
     pub fn try_recv(&mut self) -> Result<BatchStatus, oneshot::error::TryRecvError> {
         self.0.try_recv()
     }
+
+    /// Waits for receiver to resolve.
+    ///
+    /// If the sender drops before sending the batch status, returns `Err`.
+    pub async fn recv(self) -> Result<BatchStatus, oneshot::error::RecvError> {
+        self.0.await
+    }
```

Comment:
Would it make sense to change the `Output` type of the future for `BatchStatusReceiver`? Alternately, if this function is just for testing, it might make sense to mark it `#[cfg(test)]`

Reply:
I don't know how I missed the fact `BatchStatusReceiver` already had a `Future` implementation... doh!  Good eye.

Reply:
Fixed in 3bcb8e9.

Reply:
Hmm, this has different semantics, which is why I asked about changing the `Output` type. In case of an error, this remaps the error return value to `BatchStatus::Errored`, which may be the desired status for a test. In the cases this adds, it doesn't matter, so maybe it's okay.

Reply:
Yeah, I think you nailed it on the head that for how it's used here, we're fine with the subtle difference in return value.